boot/assets.go

```diff
@@ -355,7 +355,10 @@ func TrustedAssetsUpdateObserverForModel(model *asserts.Model, gadgetDir string)
 	}
 	// trusted assets need tracking only when the system is using encryption
 	// for its data partitions
-	trackTrustedAssets := hasSealedKeys(dirs.GlobalRootDir)
+	var trackTrustedAssets bool
+	if _, err := sealedKeysMethod(dirs.GlobalRootDir); err == nil {
```

Comment:
then here we could check for the special value perhaps ?

Reply:
I updated this, let me know if it's what you had in mind, I can tweak further of course.